### PR TITLE
Make sure date-range gets passed down to accessor

### PR DESCRIPTION
**Pull request by sravfeyn**
@snopoke @millerdev

### CODE REVIEW

**Comment by sravfeyn** [comment]
Also adding `--process_day_by_day` option, so that a command with big date range can be cancelled without having to worry about losing the progress

**Review by snopoke — APPROVED** [review]

File: corehq/blobs/management/commands/run_blob_migration.py [hunk 1]
```diff
@@ -77,6 +77,15 @@ def add_argument(*args, **kw):
                 "parameter. Example value: 20180109-20190109"
             ),
         )
+        add_argument(
+            '--process_day_by_day',
+            action='store_true',
+            default=False,
+            help=(
+                "Run migration for each day in the given date-range separately "
+                "to allow cancelling and resuming on any day. Only applicable with date-range option"
```
Comment (millerdev) [inline corehq/blobs/management/commands/run_blob_migration.py]: Curious why this was necessary? This may be less efficient than running it on the whole date range, and could result in some records not being migrated if date calculations are not done carefully.

The migration is already designed to be resumable, so it can be stopped and restarted repeatedly and it should pick up where it left of when restarted. If you're not seeing this behavior then there may be a bug (and we should track it down and fix it). I think this new option should be removed.

File: corehq/blobs/migrate.py [hunk 2]
```diff
@@ -422,6 +422,7 @@ def get_document_provider(self, date_range=None):
         iteration_key = self.iteration_key
         if date_range:
             (start, end) = date_range
+            self.reindexer.date_range = date_range
```
Comment (millerdev) [inline corehq/blobs/migrate.py]: Why was this necessary? It is already assigned above:

https://github.com/dimagi/commcare-hq/blob/4f44d15891fb9a915e8d541d3f0b15c385792fb3/corehq/blobs/migrate.py#L417

Would be best to only assign in one place. Maybe that one should be removed?

**Review by millerdev — COMMENTED** [review]
I would like to see some of these changes reverted.

**Comment by dannyroberts** [comment]
I think this is causing https://sentry.io/organizations/dimagi/issues/1035642700/?environment=india&project=136860&query=is%3Aunresolved&statsPeriod=14d

File: corehq/blobs/migrate.py (hunk 2 above)
Comment (sravfeyn) [reply corehq/blobs/migrate.py]: Yeah, I was worried if removing that would break something else. But looks like it shouldn't, will update in a separate PR.

File: corehq/blobs/management/commands/run_blob_migration.py (hunk 1 above)
Comment (sravfeyn) [reply corehq/blobs/management/commands/run_blob_migration.py]: I haven't tested ability to resume, but we need to be able to stop a bigger date range process and be able to split it into two separate processes when needed, I think that's not possible right now, is it?

In terms of some records not showing up, looks like `date_range` is inclusive, so `process_day_by_day` process for each day as a separate date-range, that should include all data, right?

In terms of efficiency, this provides ability to scale to multiple processes on fly which is useful on ICDS. Below is the migration process I am trying to follow.

There are around 32 CPUs that we could use, so, I am gonna split the 2018-2019 into ~32 date ranges and run them with `--process_day_by_day`. Since the data is not equally distributed on all months, some processes would finish quicker than other. Then, we can split other processes into two.  Happy to change the process, if there is a better way. (I was thinking to put all dates into a SQL table and then have processes process one date after other from the table which is more automatic, but requires more effort to get right) This was easiest.

**Comment by sravfeyn** [comment]
https://github.com/dimagi/commcare-hq/pull/24369

File: corehq/blobs/management/commands/run_blob_migration.py (hunk 1 above)
Comment (millerdev) [reply corehq/blobs/management/commands/run_blob_migration.py]: > we need to be able to stop a bigger date range process and be able to split it into two separate processes when needed, I think that's not possible right now, is it?

Interesting. You're correct, there is no support for that mode of operation right now.

> Since the data is not equally distributed on all months, some processes would finish quicker than other. Then, we can split other processes into two.

I'm concerned that the "split other processes into two" part is going to be tricky. Have you worked out a process for that part yet?

**Comment by millerdev** [comment]
nit: `--process_day_by_day` should be renamed to `--process-day-by-day` to be consistent with other options (which use dashes instead of underscores). `argparse` automatically converts dashes to underscores when it assigns the value to the options object.
